biz.aQute.bndlib/src/aQute/bnd/osgi/Builder.java

```diff
@@ -695,11 +696,9 @@ private Set<Instruction> doExpand(Jar jar, MultiMap<String, Jar> index, Instruct
 
 			Attrs directives = e.getValue();
 
-			// We can optionally filter on the
-			// source of the package. We assume
-			// they all match but this can be overridden
-			// on the instruction
-			Instruction from = new Instruction(directives.get(FROM_DIRECTIVE, "*"));
+			String fromDirective = directives.get(FROM_DIRECTIVE, "*");
+			Instruction from = new Instruction(fromDirective);
+			boolean project = fromDirective.equals(FROM_DIRECTIVE_PROJECT);
```

Comment:
This is too broad since it covers the `from` directive from more than just `-includepackage`.

Reply:
Yes, that is nice isn't it?